### PR TITLE
cleanup lucene dependency

### PR DESCRIPTION
**Pull request by ymatsuda**
@eishay @andrewconner @yingjieMiao @raymondng 

removed the dependency to Lucene from common

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/scraper/Signature.scala [hunk 1]
```diff
@@ -41,68 +25,83 @@ class Signature(val bytes: Array[Byte]) {
...
+
+    private def updateSketch(seed: Int) {
+      var v = seed.toLong & 0x7FFFFFFFFFFFFFFFL
+      var i = 0
+      val sketchSize = sketch.length
+      while (i < Signature.SIZE) {
+        v = (v * 0x5DEECE66DL + 0x123456789L) & 0x7FFFFFFFFFFFFFFFL // linear congruential generator
+        sketch(i) = min(sketch(i), (v % 0x7FFFFFFFL).toInt) // positive int
+        i += 1
+      }
```
Comment (yingjieMiao) [inline kifi-backend/modules/common/app/com/keepit/scraper/Signature.scala:104]: wow, git did a poor job in computing diff.

File: kifi-backend/project/Build.scala [hunk 2]
```diff
@@ -91,8 +91,6 @@ object ApplicationBuild extends Build {
     "org.imgscalr" % "imgscalr-lib" % "4.2",
     "us.theatr" %% "akka-quartz" % "0.2.0_42.1",
     "org.jsoup" % "jsoup" % "1.7.1",
-    "org.apache.lucene" % "lucene-core" % "4.7.0", // todo(andrew/yasuhiro): remove from common
-    "org.apache.lucene" % "lucene-analyzers-common" % "4.7.0", // todo(andrew/yasuhiro): remove from common
```
Comment (eishay) [inline kifi-backend/project/Build.scala:95]: :zap::zap::zap:

**Comment by eishay** [comment]
great!

File: kifi-backend/modules/common/app/com/keepit/scraper/Signature.scala [hunk 3]
```diff
@@ -41,68 +25,83 @@ class Signature(val bytes: Array[Byte]) {
...
+  implicit val format: Format[Signature] = Format[Signature](Reads.of[String].fmap(Signature.apply), Writes { signature => JsString(signature.toBase64()) })
 
-  private[this] val cancelerShift = windowSize % 32
+  val SIZE = 100
+  def apply(arr: Array[Byte]) = new Signature(arr)
+  def apply(base64: String) = new Signature(parseBase64Binary(base64))
+
+  def empty(): Signature = {
+    val bytes = new Array[Byte](Signature.SIZE)
+    Arrays.fill(bytes, Int.MaxValue.toByte)
```
Comment (yingjieMiao) [inline kifi-backend/modules/common/app/com/keepit/scraper/Signature.scala:41]: maybe a comment on why using Int.MaxValue instead of zero?

Comment (yingjieMiao) [inline kifi-backend/modules/common/app/com/keepit/scraper/Signature.scala:41]: because later you take min(sketch(i), ....) ?

**Comment by yingjieMiao** [comment]
lgtm

**Comment by andrewconner** [comment]
Nice!

**Comment by raymondng** [comment]
:+1:

**Pull request merged**